src/App.jsx

```diff
@@ -1,21 +1,37 @@
 import { useState } from 'react'
+import { useEffect } from 'react'
 import reactLogo from './assets/react.svg'
 import viteLogo from '/vite.svg'
 import './App.css'
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
+const useSemiPersistentState = () => {
+  const [todoList,setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
The initial state should be parsed from localStorage and falls back to an empty array [] if nothing is found.

const [todoList,setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);